modules/heat_conduction/src/materials/AnisoHeatConductionMaterial.C

```diff
@@ -10,137 +10,162 @@
-  k_y = isParamValid("thermal_conductivity_y") && (NULL != _thermal_conductivity_y_pp);
-  k_z = isParamValid("thermal_conductivity_z") && (NULL != _thermal_conductivity_z_pp);
-  if (k_x || k_y || k_z)
-  {
-    mooseError("Cannot define thermal conductivity value and Postprocessor");
-  }
 }
 
+template <bool is_ad>
 void
-AnisoHeatConductionMaterial::computeProperties()
+AnisoHeatConductionMaterialTempl<is_ad>::setDerivatives(GenericRankTwoTensor<is_ad> & prop,
+                                                        RankTwoTensor dprop_dT,
+                                                        const ADReal & ad_T)
 {
-  for (unsigned int qp(0); qp < _qrule->n_points(); ++qp)
-  {
-    (*_thermal_conductivity_x)[qp] =
-        _thermal_conductivity_x_pp ? *_thermal_conductivity_x_pp : _my_thermal_conductivity_x;
-    (*_thermal_conductivity_x_dT)[qp] = 0;
-    if (_thermal_conductivity_y)
+  unsigned int dim(_subproblem.mesh().dimension());
+  for (unsigned int i = 0; i < dim; ++i)
+    for (unsigned int j = 0; j < dim; ++j)
     {
-      (*_thermal_conductivity_y)[qp] =
-          _thermal_conductivity_y_pp ? *_thermal_conductivity_y_pp : _my_thermal_conductivity_y;
-      (*_thermal_conductivity_y_dT)[qp] = 0;
-    }
-    if (_thermal_conductivity_z)
-    {
-      (*_thermal_conductivity_z)[qp] =
-          _thermal_conductivity_z_pp ? *_thermal_conductivity_z_pp : _my_thermal_conductivity_z;
-      (*_thermal_conductivity_z_dT)[qp] = 0;
+      if (ad_T < 0)
+        prop(i, j).derivatives() = 0.0;
+      else
+        prop(i, j).derivatives() = dprop_dT(i, j) * ad_T.derivatives();
     }
+}
 
-    if (_specific_heat_temperature_function)
-    {
-      Point p;
-      _specific_heat[qp] = _specific_heat_temperature_function->value(_temperature[qp], p);
-    }
-    else
+template <>
+void
+AnisoHeatConductionMaterialTempl<false>::setDerivatives(GenericRankTwoTensor<false> &,
+                                                        RankTwoTensor,
+                                                        const ADReal &)
+{
+  mooseError("Mistaken call of setDerivatives in a non-AD AnisoHeatConductionMaterial");
+}
+
+template <bool is_ad>
+void
+AnisoHeatConductionMaterialTempl<is_ad>::setDerivatives(GenericReal<is_ad> & prop,
+                                                        Real dprop_dT,
+                                                        const ADReal & ad_T)
+{
+  if (ad_T < 0)
+    prop.derivatives() = 0;
+  else
+    prop.derivatives() = dprop_dT * ad_T.derivatives();
+}
+
+template <>
+void
+AnisoHeatConductionMaterialTempl<false>::setDerivatives(Real &, Real, const ADReal &)
+{
+  mooseError("Mistaken call of setDerivatives in a non-AD AnisoHeatConductionMaterial version");
+}
+
+template <bool is_ad>
+void
+AnisoHeatConductionMaterialTempl<is_ad>::computeQpProperties()
+{
+  Real temp_qp(0.0);
+  if (is_ad)
+    temp_qp = MetaPhysicL::raw_value(_ad_T[_qp]);
+  else
+    temp_qp = _T[_qp];
```

Comment:
```suggestion
  Real temp_qp = MetaPhysicL::raw_value(_T[_qp]);
```

`_T` wil be a `const GenericVariableValue<is_ad> &` (see comment about `coupledGenericValue` further up)

Reply:
Done